lib/src/manup_service.dart

```diff
@@ -65,13 +71,89 @@ class ManUpService {
   @visibleForTesting
   Future<Metadata> getMetadata() async {
     try {
-      var data = await this.http.get(this.url);
-      this.http.close();
-      
+      var client = this.delegate.httpClient;
```

Comment:
Unsure why this is now being passed in?

Reply:
Yes as we close the connection once we fetch the data and could not reopen connection. so we need new client every time,

Reply:
Resolving as discussion is happening elsewhere